examples/ultralytics-yolo/benchmark.py

```diff
@@ -385,18 +399,38 @@ def _load_model(args) -> (Any, bool):
         )
     elif args.engine == TORCH_ENGINE:
         print(f"loading torch model for {args.model_filepath}")
-        model = torch.load(args.model_filepath)
+        args.model_filepath = download_torch_model_if_stub(args.model_filepath)
+
+        model = torch.load(args.model_filepath, map_location=args.device)
+
         if isinstance(model, dict):
             model = model["model"]
-        model.to(args.device)
+
+        if args.recipe:
+            manager = ScheduledModifierManager.from_yaml(args.recipe)
+            model.float()  # apply recipe with float32 params
+            set_params_requires_grad(model, True)  # required to enable recipe hooks
+            manager.apply(model)
+            set_params_requires_grad(model, False)  # disable for benchmarking
+
         model.eval()
+        model.to(args.device)
+
         if args.fp16:
             print("Using half precision")
             model.half()
         else:
-            print("Using full precision")
             model.float()
+
+        if args.device == "cpu":
+            # convert any QAT models to fully quantized for CPU execution
+            try:
+                model = torch.quantization.convert(model)
+            except Exception:
+                pass
```

Comment:
Thoughts on using the following idiom?

```suggestion
            with suppress(Exception):
                model = torch.quantization.convert(model)
```

`suppress` decorator can be imported from contextlib
`from contextlib import suppress`
Available from python3.4 onwards

Reply:
![](https://media3.giphy.com/media/DeobURBiyoMRFO4GqD/giphy.gif)

Reply:
![](https://media1.giphy.com/media/QEYYlJqOaEhXrjTrOH/giphy.gif)